extensions/message-ids.md

```diff
@@ -0,0 +1,128 @@
+---
+title: IRCv3 Message IDs
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "James Wheare"
+    period: 2016
+    email: "james@irccloud.com"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `msgid` tag name. Instead, implementations SHOULD use
+the `draft/msgid` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Introduction
+
+This specification describes a mesage tag indicating a server supplied unique ID for events.
```

Comment:
Why is this limited to a server-supplied ID? The ability for clients to send a unique ID along with their message / event would be useful - it allows for the client to know the ID of the message it sent, which is required for proper processing of #288 

Rather than limit the ID supply to the server, it may be more useful to have the original sender generate the ID.